src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,197 @@
+package wooteco.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Sections {
+
+    private static final int DELETION_THRESHOLD = 1;
+
+    private final List<Section> value;
+
+    public Sections(List<Section> value) {
+        this.value = arrangeFromUpToDown(value);
+    }
+
+    private List<Section> arrangeFromUpToDown(List<Section> value) {
+        List<Section> sections = new ArrayList<>();
+        Section section = findTopSection(value);
+        sections.add(section);
+
+        while (sections.size() < value.size()) {
+            section = findNext(section, value);
+            sections.add(section);
+        }
+        return sections;
+    }
+
+    private Section findTopSection(List<Section> value) {
+        return value.stream()
+                .filter(it -> isTopSection(it, value))
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("올바르지 않은 구간들 입니다."));
+    }
+
+    private boolean isTopSection(Section section, List<Section> value) {
+        return value.stream()
+                .allMatch(it -> !it.matchDownStationWithUpStationOf(section));
```

Comment:
```suggestion
        return value.stream()
                .noneMatch(it -> it.matchDownStationWithUpStationOf(section));
```

이것도 IntelliJ 경고를 따라가보니 가독성이 더 좋아졌네요 :)

Reply:
수정했습니다!